src/plugins/saved_objects_management/server/routes/get_allowed_types.ts

```diff
@@ -17,6 +17,7 @@ export const registerGetAllowedTypesRoute = (router: IRouter) => {
     async (context, req, res) => {
       const allowedTypes = context.core.savedObjects.typeRegistry
         .getImportableAndExportableTypes()
+        .filter((type) => type.management!.visibleInManagement ?? true)
         .map((type) => type.name);
```

Comment:
Note: I could have added the same filtering in other SOM routes (e.g `find`, `scroll_count`), but I think it's not worth it because:
- The list of type from the `_allowed_types` endpoint is the source of truth for the UI regarding the types the user can search for or list in the SOM
- Even if the user was to manually call the SOM endpoint while specifying types that are exportable but not visible in management, it's not a vulnerability or a privilege escalation, as it would have been allowed to do the same using the public SO apis)